query/security/ldap/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -12,27 +12,33 @@
  **/
 -->
 <blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0">
-    <service id="ldapFieldProvider" interface="org.codice.ddf.admin.api.FieldProvider">
-        <bean class="org.codice.ddf.admin.ldap.LdapFieldProvider">
-            <argument ref="configuratorFactory"/>
-            <argument ref="featureActions"/>
-            <argument ref="managedServiceActions"/>
-            <argument ref="propertyActions"/>
-            <argument ref="serviceActions"/>
-            <property name="ldapDiscoveryFields" ref="ldapDiscoveryFields"/>
-            <property name="ldapMutationFields" ref="ldapMutationFields"/>
-        </bean>
-    </service>
 
     <reference-list id="ldapDiscoveryFields"
                     interface="org.codice.ddf.admin.api.Field"
                     availability="optional"
-                    filter="(fieldProviderId=query.ldap.*)"/>
+                    filter="(fieldProviderId=query.ldap.*)"
+                    >
+        <reference-listener bind-method="bindField" unbind-method="unbindField" ref="ldapFieldProvider"/>
+    </reference-list>
 
     <reference-list id="ldapMutationFields"
                     interface="org.codice.ddf.admin.api.fields.FunctionField"
                     availability="optional"
-                    filter="(fieldProviderId=mutation.ldap.*)"/>
+                    filter="(fieldProviderId=mutation.ldap.*)">
+        <reference-listener bind-method="bindField" unbind-method="unbindField" ref="ldapFieldProvider"/>
+    </reference-list>
+
+    <bean id="ldapFieldProvider" class="org.codice.ddf.admin.ldap.LdapFieldProvider">
+        <argument ref="configuratorFactory"/>
+        <argument ref="featureActions"/>
+        <argument ref="managedServiceActions"/>
+        <argument ref="propertyActions"/>
+        <argument ref="serviceActions"/>
+        <property name="ldapDiscoveryFields" ref="ldapDiscoveryFields"/>
+        <property name="ldapMutationFields" ref="ldapMutationFields"/>
+    </bean>
+
+    <service ref="ldapFieldProvider" interface="org.codice.ddf.admin.api.FieldProvider" />
```

Comment:
❓ Is there a reason you had to un-nest the bean from the service or is that just personal preference?